test/fixtures/increments.js

```diff
@@ -101,4 +101,19 @@ module.exports = [
   ['1.2.3-dev.bar', 'prerelease', '1.2.3-dev.0', false, 'dev', '0'],
   ['1.2.0', 'preminor', '1.3.0-dev.1', false, 'dev', '1'],
   ['1.2.3-1', 'preminor', '1.3.0-dev.0', false, 'dev'],
+
+  ['1.2.0-1', 'prerelease', '1.2.0-alpha', false, 'alpha', false],
+  ['1.2.1', 'prerelease', '1.2.2-alpha', false, 'alpha', false],
+  ['1.2.2', 'prerelease', '1.2.3-alpha', false, 'alpha', false],
+  ['1.2.0', 'prepatch', '1.2.1-dev', false, 'dev', false],
+  ['1.2.0-1', 'prepatch', '1.2.1-dev', false, 'dev', false],
+  ['1.2.0', 'premajor', '2.0.0-dev', false, 'dev', false],
+  ['1.2.3-1', 'premajor', '2.0.0-dev', false, 'dev', false],
+  ['1.2.3-dev.bar', 'prerelease', '1.2.3-dev', false, 'dev', false],
+  ['1.2.0', 'preminor', '1.3.0-dev', false, 'dev', false],
+  ['1.2.3-1', 'preminor', '1.3.0-dev', false, 'dev', false],
+  ['1.2.3-dev', 'prerelease', '1.2.3-dev.1', false, 'dev', false],
```

Comment:
my gut reaction is we should throw because they've asked for us to do something that doesn't make sense. going from no number to adding a number when we've been asked to not add numbers feels wrong

Reply:
I agree, let's throw in this situation.  No more guessing user intent!

Reply:
just to clarify do we want to throw also in:
```js
['1.2.0', 'prerelease', '1.2.1-1', false, '', false],
````
?

Reply:
Oooh good point. We sure do!  "increase the prereleas with no identifier and no identifierBase" sure sounds impossible.

Reply:
i updated it to throw on both cases